mlrun/api/api/endpoints/workflows.py

```diff
@@ -232,14 +232,11 @@ def _get_workflow_by_name(
     :param project:     MLRun project
     :param name:        workflow name
 
-    :return: workflow as a dict if project has the workflow, otherwise raises a bad request exception
+    :return: workflow as a dict if project has the workflow
     """
     for workflow in project.spec.workflows:
         if workflow["name"] == name:
             return workflow
-    log_and_raise(
-        reason=f"workflow {name} not found in project",
-    )
```

Comment:
to simplify, you can return an empty dict here instead of checking if it returned something.
also I think it should be commented why we allow a name that is not in the project workflows.